src/server/pps/server/api_server.go

```diff
@@ -651,6 +651,45 @@ func (a *apiServer) ListJobSet(request *pps.ListJobSetRequest, serv pps.API_List
 
 	// Track the jobsets we've already processed
 	seen := map[string]struct{}{}
+	// Prepare filter functions.
+	projectsFilter := make(map[string]bool, len(request.GetProjects()))
+	for _, project := range request.GetProjects() {
+		projectsFilter[project.GetName()] = true
+	}
+	jqFilter, err := jqFilterFunc(serv.Context(), request.GetJqFilter())
+	if err != nil {
+		return errors.Wrap(err, "error creating jq filter function")
+	}
+	getJobSet := func(jobInfo *pps.JobInfo) (*pps.JobSetInfo, error) {
```

Comment:
Why is this declared here, only to be used once, [60 lines later](https://github.com/pachyderm/pachyderm/pull/8864/files#diff-e400d3c661e4039b9f23301b99e8a22707cf3fdfa18865f569b499d3d708840dR726)? It would be more readable in context. I don’t believe that you’re saving any computation by declaring this outside of the closure passed at [line 710](https://github.com/pachyderm/pachyderm/pull/8864/files#diff-e400d3c661e4039b9f23301b99e8a22707cf3fdfa18865f569b499d3d708840dR710).